File: 7. Reverse Integer/reverse-integer.py
class Solution(object):
    def reverse(self, x):
        """
        :type x: int
        :rtype: int
        """
        isNegative = False
        if x < 0:
            isNegative = True
            x = -x
        reverse_x = int(str(x)[::-1])
        if reverse_x > 2**31-1 or reverse_x < -(2**31):
            return 0
        if isNegative:
            return -reverse_x
        return reverse_x

File: 7. Reverse Integer/reverse-integer.cpp
class Solution {
public:
    int reverse(int x) {
        long long rev_x = 0;
        while(x != 0){
            rev_x = x % 10 + rev_x * 10;
            x /= 10;
        }
        return rev_x < INT_MIN || rev_x > INT_MAX ? 0 : rev_x;
    }
};
